src/ModalBase/ModalCloseButton.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from '..';
+import ModalContext from './ModalContext';
+
+const ModalCloseButton = React.forwardRef(({
+  as, children, onClick, ...props
+}, ref) => {
+  const { close } = useContext(ModalContext);
+
+  return React.createElement(
+    as,
```

Comment:
I'm familiar with this pattern, but for people who aren't, you wanna throw in a comment above this line explaining WTF is going on here? 😄